packages/grpc-js/src/call-stream.ts

```diff
@@ -347,13 +358,7 @@ export class Http2CallStream extends Duplex implements Call {
        * library. In this handler we don't wait for trailers before ending the
        * call. This should ensure that this endCall happens sooner than the one
        * in the stream.on('close', ...) handler. */
-      stream.session.socket.on('close', () => {
-        this.endCall({
-          code: Status.UNAVAILABLE,
-          details: 'Connection dropped',
-          metadata: new Metadata(),
-        });
-      });
+      stream.session.socket.on('close', this.socketCloseCallback);
```

Comment:
Would a `once` listener have the same effect, or is this potentially meant to be called multiple times?

Reply:
A `once` listener should also work here, but it shouldn't make much of a different because I have to explicitly clean it up either way. And this is just a temporary solution to part of the problem.

Reply:
Fair enough. I saw there were other conditions in the `endCall` handler that had to be met in order to trigger the `removeListener` call. Although, if `once` would also work here, that seems to mean that you could also just let Node's internals handle the removal for you.

No big deal, just thought it might simplify things, but I'm not close enough to the code to know what I'm talking about :)

Reply:
The memory leak happens with connections that persist for a long time, which is also when the event will never trigger and never get automatically removed.